src/Adapter/Order/AbstractOrderHandler.php

```diff
@@ -275,4 +275,80 @@ protected function clearTemporarySpecificPrices(array $temporarySpecificPrices):
             $specificPrice->delete();
         }
     }
+
+    /**
+     * Create a specific price, or update it if it already exists
+     *
+     * @param Number $priceTaxIncluded
+     * @param Number $priceTaxExcluded
+     * @param int $productQuantity
+     * @param Order $order
+     * @param Product $product
+     * @param $combination
+     *
+     * @return SpecificPrice
+     *
+     * @throws PrestaShopException
+     * @throws \PrestaShopDatabaseException
+     */
+    protected function updateSpecificPrice(
+        Number $priceTaxIncluded,
+        Number $priceTaxExcluded,
+        int $productQuantity,
+        Order $order,
+        Product $product,
+        $combination
+    ): ?SpecificPrice {
+        $existingSpecificPrice = SpecificPrice::getSpecificPrice(
+            $product->id,
+            0,
+            $order->id_currency,
+            0,
+            0,
+            $productQuantity,
```

Comment:
I believe you don't need this `$productQuantity` parameter, it is useful when we set SpecificPrice that depends on quantity used (in the Product page). But in this case it is only used to override the catalog price, so you can use 0 as this value this should be compatible with the query in

https://github.com/PrestaShop/PrestaShop/blob/develop/classes/SpecificPrice.php#L554-L556

Because this restriction is only useful when `$specificPrice->from_quantity` is `> 1`